libpod/options.go

```diff
@@ -636,3 +637,58 @@ func WithPodLabels(labels map[string]string) PodCreateOption {
 		return nil
 	}
 }
+
+// WithDNSSearch sets the additional search domains of a container
+func WithDNSSearch(searchDomains []string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+		if StringInSlice(".", searchDomains) && len(searchDomains) > 1 {
+			return errors.Errorf("cannot pass additional search domains when also specifying '.'")
```

Comment:
Should have a constant error here EINVAL or something, so callers to the library could check.